tests/Http/Event/CopyEventRequestHandlerTest.php

```diff
@@ -0,0 +1,889 @@
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Time;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\PeriodicCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\PermanentCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SingleSubEventCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\Status;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\StatusReason;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\StatusType;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SubEvent;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SubEvents;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\TranslatedStatusReason;
+use CultuurNet\UDB3\Model\ValueObject\Translation\Language;
+use DateTimeImmutable;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidFactoryInterface;
+
+class CopyEventRequestHandlerTest extends TestCase
+{
+    use AssertApiProblemTrait;
+
+    private MockObject $uuidFactory;
+    private CallableIriGenerator $iriGenerator;
+    private TraceableCommandBus $commandBus;
+    private CopyEventRequestHandler $copyEventRequestHandler;
+
+    private const ORIGINAL_EVENT_ID = '609a8214-51c9-48c0-903f-840a4f38852f';
+    private const NEW_EVENT_ID = '581ef86b-78bf-4083-ac88-a42d0f4dde87';
+
+    protected function setUp(): void
+    {
+        $this->commandBus = new TraceableCommandBus();
+        $this->commandBus->record();
+
+        $this->uuidFactory = $this->createMock(UuidFactoryInterface::class);
+        $this->uuidFactory->expects($this->any())
+            ->method('uuid4')
+            ->willReturn(Uuid::fromString(self::NEW_EVENT_ID));
+
+        $this->iriGenerator = new CallableIriGenerator(
+            fn (string $id) => 'https://mock.io/events/' . $id
+        );
+
+        $this->copyEventRequestHandler = new CopyEventRequestHandler(
+            $this->commandBus,
+            $this->uuidFactory,
+            $this->iriGenerator
+        );
+    }
+
+    /**
+     * @test
+     * @dataProvider validEventDataProvider
+     */
+    public function it_does_not_throw_when_given_valid_event_data($data, CopyEvent $expectedCommand): void
+    {
+        $response = $this->copyEventRequestHandler->handle(
+            (new Psr7RequestBuilder())
+                ->withBodyFromString(json_encode($data))
+                ->withRouteParameter('eventId', self::ORIGINAL_EVENT_ID)
+                ->build('POST')
+        );
+
+        $expectedJson = '{"eventId":"' . self::NEW_EVENT_ID . '","url":"https:\/\/mock.io\/events\/' . self::NEW_EVENT_ID . '"}';
+
+        $this->assertEquals([$expectedCommand], $this->commandBus->getRecordedCommands());
+        $this->assertEquals(201, $response->getStatusCode());
+        $this->assertEquals($expectedJson, $response->getBody()->getContents());
+    }
+
+    public function validEventDataProvider(): array
+    {
+        return [
+            'single' => [
+                'data' => (object) [
```

Comment:
TIL it is possible to give the data itself also meaningful names and not just the scenario.

Reply:
Yeah but just by coincidence. Because the test data has to be an array, and it doesn't matter to PHPUnit if the keys are sequential numbers (normal array with only values) or strings (associative array with keys). It only uses the values. But if you make it an associative array you can add some extra meaning for easier readability.